patcher/Patcher.py

```
@@ -324,84 +238,81 @@ def align(v, a):
+    generate_hook_files(config.target_folder_path/"hooks", build_hooks_folder)
+    generate_hook_files(config.target_folder_path /
+                        "section", build_hooks_folder)
 
-    if run_system(
-            f"""cd {build_folder_path} &
-            {gcc_compiler_path} -c {GCC_FLAGS_ASM} ../hooks/*.cpp"""):
+    if run_process((
+        config.gcc_path, "-c", *config.asm_flags, build_hooks_folder / "*.cpp"
+    ), cwd=build_hooks_folder):
+        raise Exception(
+            "Errors occurred during building of generated hooks files")
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)